src/middlewared/middlewared/plugins/smb.py

```diff
@@ -1073,6 +1076,28 @@ async def query(self, filters, options):
             return await super().query(filters, options)
         return result
 
+    @private
+    async def legacy_afp_check(self, data, schema, verrors):
```

Comment:
You could just return if data['afp'] is False?

Reply:
Not really, because an existing share on the same path may be `true`

Reply:
Sorry, if True

Reply:
In that case, we may have a share on a parent or child directory that is False. :)